Cython/Utility/arrayarray.h

```diff
@@ -30,33 +30,34 @@ typedef struct arraydescr {
     char *formats;
 } arraydescr;
 
+typedef union {
+    char *ob_item;
+    float *as_floats;
+    double *as_doubles;
+    int *as_ints;
+    unsigned int *as_uints;
+    unsigned char *as_uchars;
+    signed char *as_schars;
+    char *as_chars;
+    unsigned long *as_ulongs;
+    long *as_longs;
+    unsigned long long *as_ulonglongs;
+    long long *as_longlongs;
+    short *as_shorts;
+    unsigned short *as_ushorts;
+    // Don't use Py_UNICODE ourselves in the union. This avoids deprecation warnings 
+    // for anyone who uses array.array but doesn't use this field.
+    #if PY_VERSION_HEX >= 0x030d0000
+    Py_DEPRECATED(3.13) 
+    #endif
+        wchar_t *as_pyunicodes;
+    void *as_voidptr;
+} __data_union;
```

Comment:
```suggestion
} __Pyx_data_union;
```

I'd like to to give this the standard `__Pyx_` prefix that Cython gives to almost everything internal given that we're putting this in the global namespace. (Although this file is pretty bad for not following that rule)